src/number_theory/number_field/canonical_embedding.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2022 Xavier Roblot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Xavier Roblot
+-/
+
+import number_theory.number_field.embeddings
+
+/-!
+# Canonical embedding of a number field
+The canonical embedding of a number field `K` of signature `(r₁, r₂)` is the ring homomorphism
+`K →+* ℝ^r₁ × ℂ^r₂` that sends `x ∈ K` to `(φ_₁(x),...,φ_r₁(x)) × (ψ_₁(x),..., ψ_r₂(x))` where
+`φ_₁,...,φ_r₁` are its real embeddings and `ψ_₁,..., ψ_r₂` are its complex embeddings (up to
+complex conjugation).
+
+## Main definitions and results
+* `number_field.canonical_embedding.ring_of_integers.inter_ball_finite`: the intersection of the
+image of the ring of integers by the canonical embedding and any ball centered at `0` of finite
+radius is finite.
+
+## Tags
+number field, infinite places
+-/
+
+open_locale classical number_field
+
+noncomputable theory
+
+open number_field number_field.infinite_place module fintype finite_dimensional
+
+variables (K : Type*) [field K]
+
+localized "notation `E` :=
+  ({w : infinite_place K // is_real w} → ℝ) × ({w : infinite_place K // is_complex w} → ℂ)"
+  in canonical_embedding
```

Comment:
This feels a bit weird to me. Why not give this space a proper definition, rather than just assigning to the letter E?

Reply:
Yes, you're right. It is now defined as `number_field.canonical_embedding.space`.